leshan-server-cluster/src/main/java/org/eclipse/leshan/server/cluster/RedisRegistrationStore.java

```diff
@@ -139,15 +140,20 @@ public Deregistration addRegistration(Registration registration) {
                 byte[] k = toEndpointKey(registration.getEndpoint());
                 byte[] old = j.getSet(k, serializeReg(registration));
 
-                // add registration: secondary index
-                byte[] idx = toRegIdKey(registration.getId());
-                j.set(idx, registration.getEndpoint().getBytes(UTF_8));
+                // add registration: secondary indexes
+                byte[] regid_idx = toRegIdKey(registration.getId());
+                j.set(regid_idx, registration.getEndpoint().getBytes(UTF_8));
+                byte[] addr_idx = toRegAddrKey(registration.getSocketAddress());
+                j.set(addr_idx, registration.getEndpoint().getBytes(UTF_8));
 
                 if (old != null) {
                     Registration oldRegistration = deserializeReg(old);
                     // remove old secondary index
                     if (registration.getId() != oldRegistration.getId())
```

Comment:
Not related to this PR but this string equality looks buggy...

Reply:
I will fix this in a separated commit